nodes/widgets/ui_text_input.js

```diff
@@ -1,20 +1,48 @@
 const datastore = require('../store/data.js')
+const statestore = require('../store/state.js')
+const { appendTopic } = require('../utils/index.js')
 
 module.exports = function (RED) {
     function TextInputNode (config) {
-        const node = this
-
         // create node in Node-RED
         RED.nodes.createNode(this, config)
-
-        // this ndoe need to store content/value from UI
-        node.value = null
+        const node = this
 
         // which group are we rendering this widget
         const group = RED.nodes.getNode(config.group)
 
         const evts = {
-            onChange: true,
+            beforeSend: async function (msg) {
+                const updates = msg.ui_update
+                if (updates) {
+                    if (typeof updates.label !== 'undefined') {
+                        // dynamically set "label" property
+                        statestore.set(group.getBase(), node, msg, 'label', updates.label)
+                    }
+                    if (typeof updates.mode !== 'undefined') {
+                        // dynamically set "mode" property
+                        statestore.set(group.getBase(), node, msg, 'mode', updates.mode)
+                    }
+                    if (typeof updates.clearable !== 'undefined') {
+                        // dynamically set "clearable" property
+                        statestore.set(group.getBase(), node, msg, 'clearable', updates.clearable)
+                    }
+                    if (typeof updates.icon !== 'undefined') {
+                        // dynamically set "icon" property
+                        statestore.set(group.getBase(), node, msg, 'icon', updates.icon)
+                    }
+                    if (typeof updates.iconPosition !== 'undefined') {
+                        // dynamically set "icon position" property
+                        statestore.set(group.getBase(), node, msg, 'iconPosition', updates.iconPosition)
+                    }
+                    if (typeof updates.iconInnerPosition !== 'undefined') {
+                        // dynamically set "icon inner position" property
+                        statestore.set(group.getBase(), node, msg, 'iconInnerPosition', updates.iconInnerPosition)
+                    }
+                }
+                msg = await appendTopic(RED, config, node, msg)
```

Comment:
```suggestion
```

No need for this as it's done in the `ui-base` logic for `onChange` and `onAction` already, and we have no use for `onInput` (where the `appendTopic` doesn't run by default) in a Text Input node